### DISCUSSION TITLE
GraphQL - list all branches by repository

### DISCUSSION THREAD

**Discussion by udayabaski** [community/community#24416] [Apps, API and Webhooks]
<p>Hi,</p>
<p>  The following query lists all the repo.</p>
<p>Is there any way to list branches and the last commit date?</p>
<pre><code>{
  search(query: "org:udayabaski", type: REPOSITORY, last: 100) {
    edges {
      node {
        ... on Repository{
          name
          url
          createdAt
          isArchived
          isDisabled
          updatedAt
          defaultBranchRef {
            name
          }
          branchProtectionRules(first:100) {
             edges {
              node {
                dismissesStaleReviews
              }
            }
          }
          collaborators(first:100) {
            totalCount
            edges {
              node {
                email
              }
            }
          }
          releases(first:10) {
            totalCount
            edges {
              node {
                author {
                  email
                }
                publishedAt
                tagName
                name
                isDraft
                isPrerelease
                tag {
                  id
                }
                tagName
              }
            }
          }
        }
      }
    }
  }
}
</code></pre>

**Comment by lee-dohm** [comment, marked as answer]
<p>This query should give you the information that you’re asking for:</p>
<pre><code>{
  search(query: "org:udayabaski", type: REPOSITORY, last: 100) {
    nodes {
      ... on Repository {
        nameWithOwner
        refs(first: 100, refPrefix: "refs/heads/") {
          nodes {
            name
            target {
              ... on Commit {
                oid
                committedDate
              }
            }
          }
        }
      }
    }
  }
}
</code></pre>
<p>The thing to know is that branches are a type of <a href="https://git-scm.com/book/en/v2/Git-Internals-Git-References" rel="nofollow noopener">ref</a> (short for reference), something that refers to another object in the git database.</p>
<p>I hope that helps!</p>
